spec/sidekiq/scheduler_spec.rb

```diff
@@ -40,8 +40,29 @@
         'args'  => '/tmp'
       }
 
-      expect(Sidekiq::Client).to receive(:push).with(process_parameters(config))
+      expect(Sidekiq::Client).to receive(:push) do |opts|
+        expect(opts['class']).to eq(SomeWorker)
+        expect(opts['args']).to eq(['/tmp'])
+      end
+
+      Sidekiq::Scheduler.enqueue_job(config)
+    end
+
+    it 'enqueue does not fail if it cannot constantize' do
+      # The job should be loaded, since a missing rails_env means ALL envs.
+      ENV['RAILS_ENV'] = 'production'
 
+      config = {
+        'cron'  => '* * * * *',
+        'class' => 'SomeOtherWorker',
+        'queue' => 'high',
+        'args'  => '/tmp'
+      }
+
+      expect(Sidekiq::Client).to receive(:push) do |opts|
+        expect(opts['class']).to eq('SomeOtherWorker')
+        expect(opts['args']).to eq(['/tmp'])
+      end
```

Comment:
so this is the use case... since there is one universal schedule, sometimes a scheduler that is in the context of project1 will schedule a job from project2; this will test that it won't fail.  

This project doesn't need to constantize the class for any reason except for ActiveJob, which I have another comment on.